### PR TITLE
feat: customize copy to parquet parameter

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This PR adds customized column config for parquet writer while copying table data to parquet files.

733->524

## Checklist

- [X] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Enhanced Parquet file writing with customizable column properties and improved compression settings.
  
- **Improvements**
  - Increased the default value of `max_active_window_runs` from 1 to 4, improving performance in specific scenarios.
  - Updated compaction test configurations to include detailed options for compaction type and settings.

- **Tests**
  - Adjusted test scenarios to reflect new compaction and merge mode settings, ensuring consistency and reliability.

### CODE REVIEW

File: src/common/datasource/src/file_format/parquet.rs [hunk 1]
```diff
@@ -216,6 +221,31 @@ pub async fn stream_to_parquet(
...
 
+/// Customizes per-column properties.
+fn column_wise_config(
+    mut props: WriterPropertiesBuilder,
+    schema: SchemaRef,
+) -> WriterPropertiesBuilder {
+    // Disable dictionary for timestamp column.
+    if let Some(ts_col) = schema.timestamp_column() {
+        let path = ColumnPath::new(vec![ts_col.name.clone()]);
+        props = props.set_column_dictionary_enabled(path, false);
```
Comment (evenyag) [inline src/common/datasource/src/file_format/parquet.rs:232]: Should we use `delta binary packed encoding` for timestamp column?

File: src/common/datasource/src/file_format/parquet.rs [hunk 2]
```diff
@@ -216,6 +221,31 @@ pub async fn stream_to_parquet(
...
+    // Use delta binary packed encoding for int32/int64 columns.
+    for column in schema.column_schemas() {
+        let data_type = &column.data_type;
+        if data_type == &ConcreteDataType::int64_datatype()
+            || data_type == &ConcreteDataType::int32_datatype()
+        {
+            let path = ColumnPath::new(vec![column.name.clone()]);
+            props = props.set_column_encoding(path, Encoding::DELTA_BINARY_PACKED);
+        }
+    }
```
Comment (evenyag) [inline src/common/datasource/src/file_format/parquet.rs:244]: I wonder if all int32/int64 should use delta binary packed encoding. Maybe we should only customize timestamp column.

**Review by evenyag — APPROVED** [review]

File: src/common/datasource/src/file_format/parquet.rs [hunk 3]
```diff
@@ -216,6 +220,21 @@ pub async fn stream_to_parquet(
     Ok(rows_written)
 }
 
+/// Customizes per-column properties.
+fn column_wise_config(
+    mut props: WriterPropertiesBuilder,
+    schema: SchemaRef,
+) -> WriterPropertiesBuilder {
+    // Disable dictionary for timestamp column.
```
Comment (killme2008) [inline src/common/datasource/src/file_format/parquet.rs:228]: Could you add some comments to describe why to disable dictionary for time index column?

**Review by killme2008 — APPROVED** [review]
Rest LGM

File: src/common/datasource/src/file_format/parquet.rs [hunk 4]
```diff
@@ -216,6 +220,30 @@ pub async fn stream_to_parquet(
...
+    }
+
+    for col in schema.column_schemas() {
+        if col.data_type.is_timestamp() {
+            let path = ColumnPath::new(vec![col.name.clone()]);
+            props = props
+                .set_column_dictionary_enabled(path.clone(), false)
+                .set_column_encoding(path, Encoding::DELTA_BINARY_PACKED)
+        }
+    }
```
Comment (evenyag) [inline src/common/datasource/src/file_format/parquet.rs:243]: Maybe we can remove the first `if` block?
```suggestion
    for col in schema.column_schemas() {
        if col.data_type.is_timestamp() {
            let path = ColumnPath::new(vec![col.name.clone()]);
            props = props
                .set_column_dictionary_enabled(path.clone(), false)
                .set_column_encoding(path, Encoding::DELTA_BINARY_PACKED)
        }
    }
```

**Comment by evenyag** [comment]
unit tests failed
https://github.com/GreptimeTeam/greptimedb/actions/runs/9869723840/job/27254004901?pr=4328

**Pull request merged**